client.go

```diff
@@ -133,8 +133,13 @@ func (c *Client) do(req *http.Request, v interface{}) (*Response, error) {
 			}
 
 			response.Errors = ve.Errors
-			response.Transaction = ve.Transaction
-			response.TransactionError = ve.TransactionError
+
+			// If the response object includes a TransactionError, set the
+			// transaction field on the response object and the TransactionError field.
+			if ve.TransactionError != nil {
+				response.transaction = &Transaction{TransactionError: ve.TransactionError}
```

Comment:
I didn't initially realize recurly returned the transaction error in two spots:

```xml
<error>
   <transaction_error>
   </transaction_error>
   <transaction>
      <transaction_error></transaction_error>
   </transaction>
</error>
```
We should also be returning the transaction to the user. Since `transaction_error` is now part of the `Transaction` struct in this PR, we should be able to do something like this:

```go
var ve struct {
	XMLName          xml.Name          `xml:"errors"`
	Errors           []Error           `xml:"error"`
	Transaction      *Transaction      `xml:"transaction,omitempty"`
}
```
Then assign `resp.transaction` to `ve.Transaction` -- that way `resp.Transaction` will contain the full transaction along with the `TransactionError` embedded. API users will have access to the full transaction info as well as the error.

Can you add some tests to confirm this behavior as well?